src/components/radio/RadioSelect.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { Radio } from './Radio';
+import { ChildForm } from '../childForm/ChildForm';
+
+export interface IRadioSelectProps {
+  name?: string;
+  value?: string;
+  disabled?: boolean;
+  onChange?: (value: string) => void;
+  children?: React.ReactElement<Radio>[] | React.ReactElement<ChildForm>[];
+}
+
+export class RadioSelect extends React.Component<IRadioSelectProps, any> {
+  private handleToggle(value: string) {
+    if (this.props.onChange) {
+      this.props.onChange(value);
+    }
+  }
+
+  render() {
+    const children = React.Children.map(this.props.children, (child: React.ReactElement<any>) => {
+      return React.cloneElement(child, {
+        name: child.props.name || this.props.name,
+        checked: this.props.value === child.props.value,
+        disabled: this.props.disabled,
+        onClick: (e: React.MouseEvent<HTMLElement>) => {
+          child.props.onClick && child.props.onClick(e);
+          this.handleToggle(child.props.value);
+        },
+      });
+    });
```

Comment:
I don't think we use React.Children.map nor React.cloneElement elsewhere. Have you tried with a more standard map/jsx element?

```typescript
const children = _.map(this.props.children, (child: React.ReactElement<any>) => {
return <child
  {...child.props}
  name={child.props.name || this.props.name}
  checked={this.props.value === child.props.value}
  disabled={this.props.disabled}
  onClick={(e: React.MouseEvent<HTMLElement>) => {
    child.props.onClick && child.props.onClick(e);
    this.handleToggle(child.props.value);
}} />
});
```

Reply:
J'utiliserais `Array.map` avant `_.map`, mais comme `children` n'est pas un array, je devrais le convertir avant avec un `toArray` qui existe aussi [ici](https://facebook.github.io/react/docs/react-api.html#react.children). Mais tant qu'à faire une conversion, aussi bien utiliser . Children.map directement. Pour `cloneElement`, être bon de faire un `<child {...props} prop2="value">`, j'aurais utiliser ça à la place.